Makefile: compile GEOS on Windows

otan opened the pull request:
Patch c-deps/geos with a cherry picked commit which allows the
compilation of GEOS on Windows. Update the Makefile to utilize these
changes.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/50684)
<!-- Reviewable:end -->

otan committed:
Makefile: compile GEOS on Windows

Patch c-deps/geos with a cherry picked commit which allows the
compilation of GEOS on Windows. Update the Makefile to utilize these
changes.

Release note: None

petermattis reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/50684#-:-MAkqG9F37Jzn2RYNUNZ:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

otan commented:
heh, didn't even request a review :P on the ball! thanks.

bors r=petermattis
